contracts/ethregistrar/BulkRenewal.sol

```diff
@@ -53,18 +57,22 @@ contract BulkRenewal is IBulkRenewal {
         override
     {
         ETHRegistrarController controller = getController();
-        for (uint256 i = 0; i < names.length; i++) {
+        uint256 length = names.length;
+        uint256 total;
+        for (uint256 i = 0; i < length; ) {
             IPriceOracle.Price memory price = controller.rentPrice(
                 names[i],
                 duration
             );
-            controller.renew{value: price.base + price.premium}(
-                names[i],
-                duration
-            );
+            uint256 totalPrice = price.base + price.premium;
+            controller.renew{value: totalPrice}(names[i], duration);
+            unchecked {
+                ++i;
+                total += totalPrice;
+            }
         }
         // Send any excess funds back
-        payable(msg.sender).transfer(address(this).balance);
+        payable(msg.sender).transfer(msg.value - total);
```

Comment:
Doesn't this ensure that any ETH sent accidentally is just stuck forever?

Reply:
Is it possible to send ETH here accidentally? I thought it reverts since there is no receive function.